soc/riscv/riscv-ite/it8xxx2/Kconfig.series

```
@@ -4,10 +4,7 @@
 config SOC_SERIES_RISCV32_IT8XXX2
 	bool "ITE IT8XXX2 implementation"
 	#depends on RISCV
-	# TODO:
-	# Error of can't link soft-float modules with single-float modules.
-	# No library built with -mabi=ilp32f -march=rv32iafc?
-	select CPU_HAS_FPU if RISCV_ISA_EXT_M
+	select CPU_HAS_FPU
```

> This doesn't affect existing builds because CONFIG_FPU gates use of the
FPU, and is off by default.

But some tests explicitly set `CONFIG_FPU=y` when `CONFIG_CPU_HAS_FPU=y`, and they will break when building with the Zephyr SDK.

Since we currently have no way of resolving these toolchain capabilities, it should at least filter by "not Zephyr SDK."

```suggestion
	select CPU_HAS_FPU if "$(ZEPHYR_TOOLCHAIN_VARIANT)" != "zephyr"
```

---

Neat! In that case I'll also leave it on if M is enabled, since that configuration does work.